web-app/src/InvoiceList/Filters.js

```diff
@@ -4,25 +4,44 @@ import {useHistory, useLocation} from 'react-router'
 import {useTranslation} from 'react-i18next'
 import {Accordion, Button, Card, Col, Form, FormCheck, InputGroup, Row} from 'react-bootstrap'
 import DatePicker from '../helpers/DatePicker'
-import {invoiceFormats} from '../utils/constants'
+import {invoiceFormats, orderingTypes} from '../utils/constants'
 import {formatDate, formatTime, parseTime} from '../utils/helpers'
 import {isInvoicesFilterValid, keepDigitsOnly, keepFloatCharactersOnly} from '../utils/validations'
+import {areCodeListsLoadedSelector, codeListsSelector} from '../cache/documentation/state'
+import {useDispatch, useSelector} from 'react-redux'
+import {getCodeLists} from '../cache/documentation/actions'
 
 export default ({getInvoices}) => {
   const {t} = useTranslation('common')
   const history = useHistory()
   const {pathname, search} = useLocation()
   const queryParams = new URLSearchParams(search)
 
+  const isCodeListsLoaded = useSelector(areCodeListsLoadedSelector)
+  const codeLists = useSelector(codeListsSelector)
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    if (!isCodeListsLoaded) {
+      dispatch(getCodeLists())
+    }
+  }, [dispatch, isCodeListsLoaded])
```

Comment:
we should probably not render currency filter until `isCodeListsLoaded` is `true`